src/content/en/updates/2019/06/image-support-for-async-clipboard.md

```diff
@@ -0,0 +1,302 @@
+
+# Image Support for the Async Clipboard API {: .page-title }
+
+{% include "web/_shared/contributors/thomassteiner.html" %}
+
+In Chrome&nbsp;66, we shipped the [text-focused portion](/web/updates/2018/03/clipboardapi)
+of the Asynchronous [Clipboard API](https://w3c.github.io/clipboard-apis/).
+Now in Chrome&nbsp;76, we are shipping the *image-focused* portion of the Asynchronous Clipboard
+API that will allow sites to programmatically copy and paste `image/png` images?
+
+Before we dive into what is shipping now, let’s briefly look back at how the Asynchronous Clipboard
+API works.
+If you remember the details, ⏭ skip ahead to the
+[new image-focused portion](#the_new_image-focused_portion_of_the_asynchronous_clipboard_api)
+of the API.
+
+## Recap of the Asynchronous Clipboard API
+
+### Copy: Writing text to the clipboard
+
+Text can be copied to the clipboard by calling `writeText()`.
+Since this API is asynchronous, the `writeText()` function returns a Promise
+that will be resolved or rejected depending on whether the passed text is copied successfully:
+
+```js
+const copyPageUrl = async () => {
+  try {
+    await navigator.clipboard.writeText(location.href);
+    console.log('Page URL copied to clipboard');
+  } catch (err) {
+    console.error('Failed to copy: ', err);
+  }
+};
+```
+
+### Paste: Reading text from the clipboard
+
+Much like copy, text can be read from the clipboard by calling `readText()`
+and waiting for the returned Promise to resolve with the text:
+
+```js
+const getClipboardContents = async () => {
+  try {
+    const text = await navigator.clipboard.readText();
+    console.log('Pasted content: ', text);
+  } catch (err) {
+    console.error('Failed to read clipboard contents: ', err);
+  }
+};
+```
+
+### Handling paste events
+
+Paste events can be handled by using the, surprise, `paste` event.
+It works nicely with the new asynchronous methods for reading clipboard text:
+
+```js
+document.addEventListener('paste', async (e) => {
+  e.preventDefault();
+  const text = await navigator.clipboard.readText();
+  console.log('Pasted text: ', text);
+});
+```
+
+### Security and permissions
+
+As usual, `navigator.clipboard` is only supported for pages served over HTTPS
+and to help prevent abuse, clipboard access is only allowed when a page is the active tab.
+Pages in active tabs can write to the clipboard without requesting permission,
+but reading from the clipboard always requires permission.
+
+When the Asynchronous Clipboard API was introduced,
+two new permissions for copy and paste were added to the
+[Permissions API](/web/updates/2015/04/permissions-api-for-the-web).
+The `clipboard-write` permission is granted automatically to pages when they are the active tab.
+The `clipboard-read` permission must be requested, which you can do by trying to read data from the
+clipboard.
+Attempting to read or write clipboard data will automatically prompt the user for permission
```

Comment:
> Attempting to read or write clipboard data will automatically prompt the user for permission if it has not already been granted.

@dway123 @pwnall  - this calls out read & write, but the paragraph above says that write will be automatically granted. Do you think I can delete that sentence? It looks like it's implied by the sentences above.